auto_nag/scripts/inactive_ni_pending.py

```diff
@@ -0,0 +1,184 @@
+
+
+class InactiveNeedinfoPending(BzCleaner):
+    def __init__(self):
+        super(InactiveNeedinfoPending, self).__init__()
+        self.max_actions = utils.get_config(self.name(), "max_actions", 7)
+        self.__recent_bug_limit = lmdutils.get_date_ymd("today") - relativedelta(
+            months=6
+        )
+
+    def get_max_actions(self):
+        return self.max_actions
+
+    def description(self):
+        return "Bugs with needinfo pending on inactive people"
+
+    def columns(self):
+        return ["id", "summary", "inactive_ni", "inactive_ni_count"]
+
+    def get_bugs(self, *args, **kwargs):
+        bugs = super().get_bugs(*args, **kwargs)
+        self.handle_inactive_requestee(bugs)
+
+        # Resolving https://github.com/mozilla/relman-auto-nag/issues/1300 should clean this
+        # including improve the wording in the template (i.e., "See the search query on Bugzilla").
+        self.query_url = utils.get_bz_search_url({"bug_id": ",".join(bugs.keys())})
+
+        return bugs
+
+    def handle_inactive_requestee(self, bugs):
+        requestee_bugs = {}
+        for bugid, bug in bugs.items():
+            for flag in bug["needinfo_flags"]:
+                if "requestee" not in flag:
+                    flag["requestee"] = ""
+
+                if flag["requestee"] not in requestee_bugs:
+                    requestee_bugs[flag["requestee"]] = [bugid]
+                else:
+                    requestee_bugs[flag["requestee"]].append(bugid)
+
+        user_activity = UserActivity()
+        inactive_users = user_activity.check_users(requestee_bugs.keys())
+        selected_bugs = []
+        for requestee, bugids in requestee_bugs.items():
+            if requestee in inactive_users:
+                selected_bugs.extend(bugids)
+
+        for bugid in set(bugs.keys()) - set(selected_bugs):
+            del bugs[bugid]
+
+        for bug in bugs.values():
+            bug["inactive_ni"] = [
+                {
+                    "id": flag["id"],
+                    "requestee": flag["requestee"],
+                    "requestee_status": user_activity.get_string_status(
+                        inactive_users[flag["requestee"]]
+                    ),
+                }
+                for flag in bug["needinfo_flags"]
+                if flag["requestee"] in inactive_users
+            ]
+            bug["inactive_ni_count"] = len(bug["inactive_ni"])
+            self.add_action(bug)
+
+    def is_recent_bug(self, bug):
+        return lmdutils.get_date_ymd(bug["creation_time"]) >= self.__recent_bug_limit
+
+    def add_action(self, bug):
+        users_num = len(set([flag["requestee"] for flag in bug["inactive_ni"]]))
+
+        if (
+            (
+                bug["priority"] in HIGH_PRIORITY
+                or bug["severity"] in HIGH_SEVERITY
+                or self.is_recent_bug(bug)
+            )
+            and not utils.is_no_assignee(bug["triage_owner"])
```

Comment:
With the current implementation, in case of bugs in a component with no triage owner, we will simply clear the needinfo request without redirecting.
We should instead not do anything, and hopefully by the next run of the tool a triage owner will be defined.

Reply:
Very good point, we shouldn't perform an action if we don't have an active triage owner. Done in ced6005998544ae7e717c1d0cd723ebb281c9cf8.